[ios] About screen redesign

kirylkaveryn opened the pull request:
Closes https://github.com/organicmaps/organicmaps/issues/7194

This PR implements the About screen redesing according to the design but with some improvements.

There are some features that are still not implemented:
1. strings localizations for this labels :
<img width="200" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/9ce12cea-617c-420e-968b-e485e38c1a86">
<img width="200" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/45ff543c-bb51-4e98-99d6-f4b34c201f2b">


Tested: ios12.5 (iPhone 6),  ios17 (iPhone 11pro, iPhone15pro sim)

### Result:
<img width="463" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/9dd5562c-1330-4c27-9061-f551ff9f13c3">
<img width="484" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/7be80513-90d5-4a51-9248-599b76a4081b">


![Simulator Screen Recording - iPhone 15 Pro - 2024-02-08 at 16 49 40](https://github.com/organicmaps/organicmaps/assets/79797627/10570170-2319-4a89-b862-3d862ee917cc)
![Simulator Screen Recording - iPhone 15 Pro - 2024-02-08 at 16 49 52](https://github.com/organicmaps/organicmaps/assets/79797627/bd5a677a-181a-4225-8b12-087d38f0d1bd)

matheusgomesms commented:
Looks great! Instagram logo is different from the rest (B&W version). I would use the original version or use all of them black and white (at least in the dark version). Also, the logo sizes look a bit too much for me, but this is my personal preference.

Also, I know this is the simplified OSM logo version (which can be used in this case), but not sure how the original logo would work. Perhaps, even with the small size, the original OSM logo version would be better to inform users?

euf commented:
Hello! I like this design very much. It looks amazing 🔥 
Can I suggest a few tweaks for text and icons? 

> Community developed maps and app
`Community-created maps and app`?

> Fast ad minimalist
There's a typo, should be `and` I guess?

> Community-created OpenStreetMap data as of OpenStreetMap data: 2 February 2024. 
Excessive wording, maybe just `OpenStreetMap data as of 2 February 2024 `?

Also I support the idea of unifying all brand icons to be black and white. Maybe source all of them from FontAwesome, since we already list them in credits?

Volunteer icon seems to differ in style due to think strokes. Maybe use [helping hands](https://fontawesome.com/icons/handshake-angle?f=classic&s=solid) icon, also from FontAwesome? 

Colored OpenStreetMap icon in dark mode becomes illegible, the magnifying glass hand blends with the background. Maybe add a white outline to the image?

kirylkaveryn commented:
> Instagram logo is different from the rest (B&W version)

Thanks a lot! I'll replace it with colored one.


> or use all of them black and white

> Also I support the idea of unifying all brand icons to be monochrome.

Using monochrome icons looks more minimalistic and stylish but doesn't simplify UX. Color is more suitable for visual navigation rather than shape. And we don't have the same `monochrome` UI elements in the whole app so for me it makes this screen a little bit weird in comparison with the rest of the OM app.

> Also, I know this is the simplified OSM logo version (which can be used in this case), but not sure how the original logo would work

Can you please take a screenshot of this original logo? I can't find it...

> Colored OpenStreetMap icon in dark mode becomes illegible, the magnifying glass hand blends with the background. Maybe add a white outline to the image?

Thanks! I will try to fix that.

kirylkaveryn commented:
Some updates were added:
1. OSM logo changed to default
2. The section with info "No Ads, No Tracking..." changed from V stack into H stack because there is no good solution to handle all devices/localizations properly and display a lot of text on "3 rows" without decreasing fonts... Its too much text to place it in a such way. It looks not so good now, but we need another UI solution to handle a lot of text.
3. Because of p2 I use previously localized text as on Android.
4. Spacing between social media increased from 16 to 25 so icons looks no to big
5. BW Instagram icon replaced with colored

@matheusgomesms @euf @rtsisyk @biodranik @oleg-PK  can you check please?

I didn't replace Volunteer icon for now, maybe we can try to find another one...
![image](https://github.com/organicmaps/organicmaps/assets/79797627/574356e9-5a60-4193-8ab2-fda49355a7ff)


## Temporary results:
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/93206676-ee9a-482d-ac6b-3024348ee359">
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/5e090d19-fa00-476d-beb0-813dbcd989a9">
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/fb1b350a-97d7-4e94-b981-c01df1c9a9f8">
<img width="719" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/82987cab-2b0c-4c83-ac37-20df465f76ec">

biodranik commented:
Thanks! Looks ok, horizontal feature icons definitely won't fit into one line on smaller screens:

![выява](https://github.com/organicmaps/organicmaps/assets/170263/9f931426-45c0-46a9-8cec-21e18a1268dd)

1. Let's try to change the title from About Organic Maps to "Organic Maps version 2024.02.11-3" and remove the version under the icon.
2. Can the "Free for everyone..." text size be automatically shrunk if necessary to fit one line?
3. It is not clear that the dialog can be scrolled... Any ideas on how it can be hinted, at least once, to teach users to scroll down?
4. The date in OSM string should be localized using `dateFormatter.dateStyle = .long`
5. Can the text size in features and OSM be slightly reduced, and the right margin/indent also reduced a bit, to fit more text (ideally in one line)? Icons can likely be reduced too if necessary. Maybe these changes should be done only for smaller screens?
6. "No battery drain, works offline"
7. "Fast, minimalist, developed by community"
8. "... Learn more about how to ..."
9. "Report or fix incorrect map data"
10. "Volunteer" (use ru for translations, or "To Volunteer", to get correct results)
11. "Rate the app to motivate us!"
12. Let's try to add a header to social icons: "Follow and contact us:"

kirylkaveryn commented:
@biodranik 
1. fixed - also make it copyable by tap and suitable for a screen wide if needed (for small devices)
2. done
3. The scroll indicator is enabled
4. I am using the previous formatting with the `long` format:
```swift
  // Returns a human-readable maps data version.
  static func formattedMapsDataVersion() -> String {
    // First, convert version code like 220131 to a date.
    let df = DateFormatter()
    df.locale = Locale(identifier:"en_US_POSIX")
    df.dateFormat = "yyMMdd"
    let mapsVersionInt = FrameworkHelper.dataVersion()
    let mapsDate = df.date(from: String(mapsVersionInt))!
    // Second, print the date in the local user's format.
    df.locale = Locale.current
    df.dateStyle = .long // < ---
    df.timeStyle = .none
    return df.string(from:mapsDate)
  }
```
5. font and image sizes are already very small...
6. ...12. fixed!

## TODO:
1. Links to:
- Report a Map Data Problem - https://organicmaps.app/faq
- Volunteer - https://organicmaps.app/volunteering
are still not working.
2. Fix translations
3. Fix in android (after the PR approving):

> changes in Android are needed, because "osm_presentation" localized text was changed from:
Cartographic data from OpenStreetMap as of %@. OSM is like Wikipedia for maps, where you can add and edit places for users around the world.
to
 Community-created OpenStreetMap data as of %@. Learn more how to edit and update the map at %@
(2 attributes instead of 1)


## Screenshots:
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/f0b76ca9-3f03-4628-8e7c-be7483ceb8ff">
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/e160a966-0f4c-44a3-b2c3-5ba214f6133e">
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/b7a8084d-d383-4ad3-8721-4f96bd8fc364">
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/ff4ef00f-cce3-4a12-a90f-3f6118d09f4a">
![Simulator Screen Recording - iPhone 15 Pro - 2024-02-11 at 20 05 34](https://github.com/organicmaps/organicmaps/assets/79797627/6d274fda-c43b-4736-92de-adc151cfeeab)

euf commented:
I checked this PR in the Simulator and I've got several more questions, if you don't mind:

- For some reason I don't see the "Donate section" 🤔 
- Opening `https://www.openstreetmap.org/` in a mobile browser leads to confusion. Let's navigate to `https://www.openstreetmap.org/help` on click instead, it would be more helpful for newcomers 
- Consider keeping new "three lines" design on iPhone and use previous "three columns" design on iPad forthe additionalInfoView? 
- Some of the new icons were drawn by you, judging from `ic_social_media_mail.svg` metadata. Others seem to be reused from somewhere else and come in pdf already. As far as I can tell, most of the currently used OM interface icons are outline icons with 2 px stroke width on a 24×24 px canvas with 0–2 px margins for optical compensation. Do you need any help with finding / drawing more icons for this screen? Would it be okay if I suggested keeping SVGs instead of PDFs whenever possible? Also I guess the reason for removing `newspaper_white_24dp.svg` was to move away from material style on iOS, is that right? 
- The volunteer icon definitely needs to be replaced with something else. Current "three raised hand in circle" icon just differs too much in visual style from its neighbors. OSM bw icon is inconsistent too, but there are no replacements. Volunteer icon can be replaced.
- First three icons (faq, report_a_bug, report_incorrect_map_bug) differ in color from last three icons (volunteer, news, rate_the_app). That's super weird, I checked twice with different color pickers, and they are not the same. Can it be fixed?
- "Follow and contact us" section has the iOS "Mail.app" logo mixed in with social network logos. I see the idea, but it looks bad placed next to LinkedIn icon, being a smaller square with more rounded corners. Maybe change its size or reorder icons to put mail away from other squares?  

Sorry for all this nitpicking, I really like your job and I don't want to sound unwelcoming. It's just a rare chance to do things properly in one go. Thanks again for doing this! 🔥

biodranik commented:
@euf thanks for the review! Can you please help with icons?

kirylkaveryn commented:
@euf thanks for your attention! I appreciate your help! Because I'm not a designer :) And usually, I use  Apple's SFSymbols for all kinds of icons, but now I'm a bit struggling :)
1. if you don't see Donate section means that it is disabled for now. I don't add this section if settings return nil (like was done before)
```swift
   if isDonateEnabled() {
      stackView.addArrangedSubviewWithSeparator(donationView)
    }
  func isDonateEnabled() -> Bool {
    return Settings.donateUrl() != nil
  }
```
@biodranik does this capability depend on the build version or smth other? On my devise, it works properly...
2. I'll fix that! Thanks!
3. For the current implementation decided not to add a new version for iPad (agreed with @biodranik ) because of a lot of additional work with string translating (these strings should be split, translated and combined in other way and it may cause additional work in Android too) and layouting should be done. And there are still some devices (such as iPhone 5, 5C, 6 which support ios 12) that in landscape orientation cannot fit all this text properly in some "long" languages. This solution is not properly designed to be implemented for now...
4. Some icons were taken from Figma's file referenced in the issue and some were from other sources (please see design in the issue https://github.com/organicmaps/organicmaps/issues/7194) 
5. PDF is a more stable vector format than SVG in iOS development. Sometimes uikit struggles with proper rendering shadows and tinting when the `template` rendering mode is selected for SVG.
6. `newspaper_white_24dp.svg` was removed because this icon isn't used anywhere in the project...
7. Please, share your ideas with the `volunteer icon`!
8. About icon colors... it is strange... because they have the same tint color on my devices - this is color picking from the screenshot:
<img width="587" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/273444c6-75dd-4a6e-b42c-c07fc6495efe">
<img width="633" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/2c1f0b5d-ddcc-4157-8361-464d7540abee">
But I will check it one more time.
9. The idea was to take icons "as they are" and not to place them in equal rectangles or circles. Making designs in the figma is not my best :)  I'll try to do smth with that.
10. Thanks! Don't know about it! I'll add this into the `view did appear`.

biodranik commented:
@rtsisyk @kirylkaveryn AZ translation is incorrect for some reason. How did you get it? Google shows:
```
Hər kəs üçün pulsuz, sevgi ilə hazırlanmışdır
```

biodranik commented:
```
/home/runner/work/organicmaps/organicmaps/android/app/src/main/java/app/organicmaps/help/HelpFragment.java:52: Error: Wrong argument count, format string osm_presentation requires 2 but format call supplies 1 [StringFormatMatches]
      osmPresentationView.setText(getString(R.string.osm_presentation, dataVersion));
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /home/runner/work/organicmaps/organicmaps/android/app/src/main/res/values-ar/strings.xml:493: This definition requires 2 arguments

```

euf commented:
Here are the resulting icons after fixes: [about-icons.zip](https://github.com/organicmaps/organicmaps/files/14337292/about-icons.zip)


| svg file | mdpi raster |
|----|----|
| ![1-faq](https://github.com/organicmaps/organicmaps/assets/11754264/92714f67-91ac-4904-8e1e-e33f124f72d4) | ![1-faq](https://github.com/organicmaps/organicmaps/assets/11754264/5c0648e1-ece4-41e4-8f99-ace7c717c60d) |
| ![2-mail](https://github.com/organicmaps/organicmaps/assets/11754264/2d5f6c1f-a01f-48d4-81ad-5908eb953008) | ![2-mail](https://github.com/organicmaps/organicmaps/assets/11754264/e29e79f8-2fd7-4e52-a778-f49db844e16f) |
| ![3-error](https://github.com/organicmaps/organicmaps/assets/11754264/1bda0978-d8b8-4b6b-9dca-0b3fad644bd5) | ![3-error](https://github.com/organicmaps/organicmaps/assets/11754264/2ceca533-d853-4d5d-99e3-e53081c08a52) |
| ![4-volunteer](https://github.com/organicmaps/organicmaps/assets/11754264/43e9cadb-024b-465c-923f-d5e1ec7c1027) | ![4-volunteer](https://github.com/organicmaps/organicmaps/assets/11754264/6108c101-4b32-47f5-9fb5-59f9047f5c0f) |
| ![5-news](https://github.com/organicmaps/organicmaps/assets/11754264/77ba6af1-bd98-44eb-9fd2-0ad6e6c85944) | ![5-news](https://github.com/organicmaps/organicmaps/assets/11754264/5b99c57d-1e76-4dbd-af96-0185aeb4f31f) |
| ![6-rate](https://github.com/organicmaps/organicmaps/assets/11754264/ba9fdd2c-40af-40b5-88d8-e05fc3e564f3) | ![6-rate](https://github.com/organicmaps/organicmaps/assets/11754264/e12320ac-3f68-4c65-bd65-70c147e5ccb8) | 


And that's how they should look: 

<img width="281" alt="Frame 23" src="https://github.com/organicmaps/organicmaps/assets/11754264/b1350a19-bd8c-406c-afb1-d3621b3445c7">

biodranik commented:
@euf thanks! What about the night mode?

@kirylkaveryn why the Android check is failing?

kirylkaveryn commented:
> why the Android check is failing?

@biodranik this is because we updated one localized string and added the OpenStreetmaps.com link as a parameter in the previous realization, but now I just place it into the string itself rather than the 2nd parameter. I'll update is soon without fixi

> @euf thanks! What about the night mode?
This is svg so I'll tint them.

kirylkaveryn commented:
@biodranik @rtsisyk @oleg-PK 

The icons were updated - thanks to @euf !
The issue with strings on the android was fixed.

## Screenshots:
### iOS 17.2 iPhone 15Pro sim
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/509f0ca8-a376-4d20-bea4-59b6b7541110">

![Simulator Screen Recording - iPhone 15 Pro - 2024-02-20 at 14 49 08](https://github.com/organicmaps/organicmaps/assets/79797627/62a2ad5e-e513-460e-ab03-6eaacdaa5701)

### iOS 17.2 iPhone 11Pro device 
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/2e5e4c77-9cf3-4c35-95de-6b3e229f659d">

### iOS 17.2 iPad 5air sim

<img width="600" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/71f8c1b7-f5e3-49a6-aa0a-7b396cf0b6f5">
<img width="600" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/8f023c96-d2e0-4926-ad51-84b87ec7ba54">

### iOS 12.5 iPhone 6 device
 
<img src="https://github.com/organicmaps/organicmaps/assets/79797627/c9563061-b3ef-4a4d-826d-8306d68df1e5" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/79797627/c2647981-24fa-4daf-9d6b-36d5203a87ba" width="300">
<img src="https://github.com/organicmaps/organicmaps/assets/79797627/3fd537e8-ee35-4bf9-8da8-37ba3c343e5b" width="350">
<img src="https://github.com/organicmaps/organicmaps/assets/79797627/0197adf2-a746-4237-8473-13a7670cdaf5" width="350">

biodranik commented:
0. The first list of features is visually the same as the clickable list of buttons below. Are there any ideas on how to improve it? CC @euf 
1. FAQ doesn't work.
2. Change the order of buttons: FAQ, map bugs, app bugs, news, volunteer, rate.
3. Fix text, should be "... Learn more about how to ..."
4. Images at the bottom are too big, maybe adding some padding on the left and right of the list would look better (not a blocker)?

<img width="1450" alt="image" src="https://github.com/organicmaps/organicmaps/assets/170263/56d54670-14ad-4e40-bbc2-d9fcbc89e889">

kirylkaveryn commented:
@biodranik I updated the feature list to visually match with the Android:
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/52385a0f-d6e9-4518-9669-d51ff005b4ac">
<img width="300" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/2c7f6c04-c2f8-474d-acf4-ce58274b36e3">

biodranik commented on iphone/Maps/UI/Help/AboutController/AboutController.swift:
```diff
@@ -0,0 +1,535 @@
+    let mapsDate = df.date(from: String(mapsVersionInt))!
+    // Second, print the date in the local user's format.
+    df.locale = Locale.current
+    df.dateStyle = .long
```
On the screenshot "28 February 2024" is displayed. In swift playground I seen "February 28, 2024" for .long (and it looks/reads better for native speakers). Is it because df.dateFormat was not reset above?

kirylkaveryn replied:
Yes! It is because `Locale.current` defined not by language, but by the user region settings and may vary.
And reset previous `df.locale = Locale(identifier:"en_US_POSIX")`

Overall it is better to use the `current`, because `en_US` and `en_UK` have different standards of writing dates:
en_UK :  2 February 2024
en_US: February 2, 2024 

So there are different native speakers even for the same language and it better to use different formats.

biodranik replied:
Aha, so you have en_UK locale set. Then its ok )

biodranik commented on iphone/Maps/UI/Help/AboutController/AboutController.swift:
```diff
@@ -0,0 +1,535 @@
+  func showCopyright() {
+    let path = Bundle.main.path(forResource: "copyright", ofType: "html")!
+    let html = try! String(contentsOfFile: path, encoding: String.Encoding.utf8)
+    let webViewController = WebViewController.init(html: html, baseUrl: nil, title: L("copyright"))!
```
Does it properly work with the dark theme?

kirylkaveryn replied:
Yes!
<img width="350" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/8e8568dd-f09d-4261-a141-c956fb10d7d8">

biodranik commented on iphone/Maps/UI/Help/AboutController/Models/AboutInfo.swift:
```diff
@@ -0,0 +1,74 @@
+enum AboutInfo: CaseIterable {
```
Why CaseIterable is needed?

kirylkaveryn replied:
Thanks! I used this one to get allCases previously.

biodranik commented on iphone/Maps/UI/Help/AboutController/Models/AboutInfo.swift:
```diff
@@ -0,0 +1,74 @@
+
+  var title: String {
+    switch self {
+
+
```
```suggestion
```

biodranik commented on iphone/Maps/UI/Help/AboutController/Models/SocialMedia.swift:
```diff
@@ -0,0 +1,62 @@
+enum SocialMedia: CaseIterable {
```
ditto

biodranik commented on iphone/Maps/UI/Help/AboutController/Models/SocialMedia.swift:
```diff
@@ -0,0 +1,62 @@
+  case organicMapsEmail
+  case github
+
+  var link: String? {
```
Doesn't `var` imply that it can also be set, not only get?

biodranik reviewed (commented):
Thanks! I like the idea of removing confusing icons. Dots look less confusing and less distracting.

kirylkaveryn commented on iphone/Maps/UI/Help/AboutController/AboutController.swift:
```diff
@@ -0,0 +1,535 @@
+        case .faq:
+          self?.navigationController?.pushViewController(FaqController(), animated: true)
+        case .reportABug:
+          guard let link = aboutInfo.link else { return }
```
This is only for the optional unwrapping because `AboutInfo`'s prop `link: String?`. Some cases have link, and some don't.
Maybe fatalError will be better here?

biodranik replied:
If issues can be discovered and fixed during the development and testing, then it's better to crash in the unexpected case, of course.
